tools/integration/test/integration/testConfig.js

```diff
@@ -32,10 +34,47 @@ const components = [
   // 'sourcearchive/mavencentral/org.apache.httpcomponents/httpcore/4.1' // Dev and prod have different license and scores. See https://github.com/clearlydefined/crawler/issues/533
 ]
 
+function shouldUseDynamicComponents() {
+  // check for environment variable DYNAMIC_COORDINATES, if it is set to true, use dynamic components
+  return process.env.DYNAMIC_COORDINATES === 'true';
+  
+}
+
+async function getComponents() {
+  if (shouldUseDynamicComponents()) {
+    console.info("Using dynamic components");
+    return componentsDynamic();
+  } else {
+    console.info("Using static components");
+    return Promise.resolve(componentsStatic);
+  }
+}
+
+const componentsDynamic = async () => {
+  
+  const filePath = path.join(__dirname, 'recentDefinitions.json');
+
+  try {
+    // Check if the file exists
+    await fs.access(filePath);
+    // Read the file contents
+    const data = await fs.readFile(filePath, 'utf8');
+    console.info("Read dynamic components from disk")
+    return JSON.parse(data);
+  } catch (err) {
+    // If the file doesn't exist, fetch the data and save it to disk
+    const response = await fetch('https://cosmos-query-function-app.azurewebsites.net/api/getrecentdefinitions?days=1&limit=1');
```

Comment:
Another source of recent harvested coordinates exists at the status endpoint: https://dev-api.clearlydefined.io/status/recentlycrawled.  The response format is: 
```
[
	{
		"coordinates": "go/golang/github.com%2Fazure/azure-sdk-for-go/v43.3.0+incompatible",
		"timestamp": "2024-09-25T22:36:22.017Z"
	},
	{
		"coordinates": "go/golang/github.com%2Fazure%2Fgo-autorest/autorest/v0.11.24",
		"timestamp": "2024-09-25T22:10:52.752Z"
	}
]
```
The internal logic is at service/statusService.  It utilize application insight, so might be cheaper than cosmo query?

Reply:
Thanks for the info, I wasn't aware of this endpoint. However it's not giving us enough data as you pointed out, so we'll have to rely on some other mechanism.

I think querying the CosmosDB should not increase the cost much if at all, because it's only done rarely and touches a limited amount of data.

Reply:
The approach looks good.  Another idea just occurred to me: the search query in the getRecentDefinition is based on _meta.updated, which is also the change publication based off.  The changed coordinates published hourly can potentially be used to provide the recent coordinates by day and by type (through sorting).  Just thought to mention it as an idea.

Reply:
Yes DB query is more extensible. Thanks for the explanation and clarification!